src/node.cc

```diff
@@ -885,8 +885,25 @@ int Start(int argc, char** argv) {
   }
 
   {
-    NodeMainInstance main_instance(
-        uv_default_loop(), result.args, result.exec_args);
+    Isolate::CreateParams params;
```

Comment:
Why not move this bit into the `NodeMainInstance::main_instance` overload? Sine 2 out of 3 params are anyway calculated using `NodeMainInstance::` static methods, and the third is `{nullptr}`;
i.e. something like:
```c++
    NodeMainInstance main_instance({nullptr},
                                   uv_default_loop(),
                                   per_process::v8_platform.Platform(),
                                   result.args,
                                   result.exec_args);
...
    NodeMainInstance main_instance(const std::vector<intptr_t> external_references&&,
                                   uv_loop_t* event_loop,
                                   MultiIsolatePlatform* platform,
                                   const std::vector<std::string>& args,
                                   const std::vector<std::string>& exec_args);
      Isolate::CreateParams params;
      v8::StartupData* blob = NodeMainInstance::GetEmbeddedSnapshotBlob();
      const NodeMainInstance::IndexArray* indexes =
          NodeMainInstance::GetIsolateDataIndexes();
      if (blob != nullptr) {
        params.external_references = external_references.data();
        params.snapshot_blob = blob;
      }
      ...
```

Reply:
I was thinking about moving the `GetEmbeddedSnapshotBlob()` etc. out of `NodeMainInstance` at some point when we add support for other snapshots, e.g. workers, and maybe restructuring the class somehow so we could reuse it in the cctest, it would be easier to experiment if the deserialization can be switched off by the caller even when the binary is built with embedded snapshot.